doc/src/modules/physics/continuum_mechanics/beam_problems.rst

```diff
@@ -451,3 +452,340 @@ away from that end. Also a clockwise moment of 5 KN-m is applied at the overhang
+  \\\\|----------|-----------|----------|-----------|
+           l           l          l            l     
+
+.. note::
+
+    Using the sign convention of upward forces and clockwise moment
+    being positive.
+
+.. code:: pycon
+
+    >>> from sympy.physics.continuum_mechanics.beam import Beam
+    >>> from sympy import symbols
+    >>> E, I = symbols('E I')
+    >>> l = symbols('l', positive=True)
+    >>> R1, M1, R2, R3, P = symbols('R1 M1 R2 R3 P')
+    >>> b1 = Beam(2*l, E, I)
+    >>> b2 = Beam(2*l, E, I)
+    >>> b = b1.join(b2, "hinge")
+    >>> b.apply_load(M1, 0, -2)
+    >>> b.apply_load(R1, 0, -1)
+    >>> b.apply_load(R2, l, -1)
+    >>> b.apply_load(R3, 4*l, -1)
+    >>> b.apply_load(P, 3*l, -1)
+    >>> b.bc_slope = [(0, 0)]
+    >>> b.bc_deflection = [(0, 0), (l, 0), (4*l, 0)]
+    >>> b.solve_for_reaction_loads(M1, R1, R2, R3)
+    >>> b.reaction_loads
+    ⎧    -P⋅l       3⋅P      -5⋅P       -P ⎫
+    ⎨M₁: ─────, R₁: ───, R₂: ─────, R₃: ───⎬
+    ⎩      4         4         4         2 ⎭
+
+    >>> b.load
+             -2          -1               -1                                -1
+      P⋅l⋅<x>     3⋅P⋅<x>     5⋅P⋅<-l + x>                 -1   P⋅<-4⋅l + x>  
+    - ───────── + ───────── - ────────────── + P⋅<-3⋅l + x>   - ──────────────
+          4           4             4                                 2       
+
+    >>> b.shear_force()
+             -1          0               0                               0
+      P⋅l⋅<x>     3⋅P⋅<x>    5⋅P⋅<-l + x>                0   P⋅<-4⋅l + x> 
+    - ───────── + ──────── - ───────────── + P⋅<-3⋅l + x>  - ─────────────
+          4          4             4                               2  
+
+    >>> b.bending_moment()
+            0          1               1                               1
+      P⋅l⋅<x>    3⋅P⋅<x>    5⋅P⋅<-l + x>                1   P⋅<-4⋅l + x> 
+    - ──────── + ──────── - ───────────── + P⋅<-3⋅l + x>  - ─────────────
+         4          4             4                               2  
+
+    >>> b.slope()
+    ⎧         1          2               2               2              
+    ⎪  P⋅l⋅<x>    3⋅P⋅<x>    5⋅P⋅<-l + x>    P⋅<-2⋅l + x>               
+    ⎪- ──────── + ──────── - ───────────── + ─────────────              
+    ⎪     4          8             8               4                    
+    ⎪─────────────────────────────────────────────────────   for x ≤ 2⋅l
+    ⎪                         E⋅I                                       
+    ⎨                                                                   
+    ⎪     2               2               2               2             
+    ⎪5⋅P⋅l    P⋅<-2⋅l + x>    P⋅<-3⋅l + x>    P⋅<-4⋅l + x>              
+    ⎪────── - ───────────── + ───────────── - ─────────────             
+    ⎪  48           4               2               4                   
+    ⎪──────────────────────────────────────────────────────  for x < 4⋅l
+    ⎩                         E⋅I                                       
```

Comment:
Why doesn't this output a single expression with only singularity functions? We've used singularity functions to specifically avoid having Piecewise in the beam results. It'd be nice to retain that, instead of mixing the two equivalent types.

Reply:
I used `SingularityFunction` first for composite beams as it is being used to evaluate these expressions, but the result was hard to read. Like for the above example, it would be;
```
⎛     2               2               2               2⎞               ⎛      
⎜5⋅P⋅l    P⋅<-2⋅l + x>    P⋅<-3⋅l + x>    P⋅<-4⋅l + x> ⎟           0   ⎜  P⋅l⋅
⎜────── - ───────────── + ───────────── - ─────────────⎟⋅<-2⋅l + x>    ⎜- ────
⎝  48           4               2               4      ⎠               ⎝     4
──────────────────────────────────────────────────────────────────── + ───────
                                E⋅I                                           

   1          2               2               2⎞        ⎛         1          2
<x>    3⋅P⋅<x>    5⋅P⋅<-l + x>    P⋅<-2⋅l + x> ⎟    0   ⎜  P⋅l⋅<x>    3⋅P⋅<x> 
──── + ──────── - ───────────── + ─────────────⎟⋅<x>    ⎜- ──────── + ────────
          8             8               4      ⎠        ⎝     4          8    
───────────────────────────────────────────────────── - ──────────────────────
                     E⋅I                                                      

               2               2⎞            
   5⋅P⋅<-l + x>    P⋅<-2⋅l + x> ⎟           0
 - ───────────── + ─────────────⎟⋅<-2⋅l + x> 
         8               4      ⎠            
─────────────────────────────────────────────

```
So, I thought it would be better to show expression for each sub-beam separately for better  readability.

Reply:
should I change `Pieceiwse` to above representation?

Reply:
Should I change Pieceiwse to above representation? What do you think?